### PR TITLE
[FLINK-2775] [CLEANUP] Cleanup code as part of theme to be more consistent on Utils classes

### PR DESCRIPTION
**Pull request by hsaputra**
As part of continuing theme and effort to help make the code more consistent, adding cleanup to Utils classes:
-) Add final class modifier to the XXXUtils and XXXUtil classes to make sure can not be extended.
-) Add missing Javadoc header classs to some public classes.
-) Add private constructor to Utils classes to avoid instantiation.
-) Remove unused test/java/org/apache/flink/test/recordJobs/util/ConfigUtils.java class

### CODE REVIEW

**Comment by hsaputra** [comment]
Looks like fail on unrelated Kafka partition error

File: flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/utils/ExampleUtils.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ public void configure(Configuration parameters) {
 				new FlatMapFunction<Long, Edge<Long, NullValue>>() {
 					@Override
 					public void flatMap(Long key,
-							Collector<Edge<Long, NullValue>> out)
+															Collector<Edge<Long, NullValue>> out)
```
Comment (zentol) [inline flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/utils/ExampleUtils.java]: formatting

File: flink-runtime/src/main/java/org/apache/flink/runtime/util/ClassLoaderUtil.java [hunk 2]
```diff
@@ -129,4 +129,11 @@ public static boolean validateClassLoadable(ClassNotFoundException cnfe, ClassLo
 			return false;
 		}
 	}
+
+	/**
+	 * No instantiation.
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/util/ClassLoaderUtil.java]: different javadoc compared to other private constructors

File: flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/util/ClusterUtil.java [hunk 3]
```diff
@@ -26,11 +26,17 @@
...
-public class ClusterUtil {
+/**
+ * Utility class to manage mini cluster for Apache Flink.
+ */
+public final class ClusterUtil {
 	
 	private static final Logger LOG = LoggerFactory.getLogger(ClusterUtil.class);
 
+	private ClusterUtil() {
+		// Prevent instantiation.
```
Comment (zentol) [inline flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/util/ClusterUtil.java]: no javadoc, doesn't throw a runtime exception as other constructors.

**Comment by StephanEwen** [comment]
Looks good. @zentol's comments are nice-to-address issues.

+1 to merge with addressing the comments

File: flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/utils/ExampleUtils.java (hunk 1 above)
Comment (hsaputra) [inline flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/utils/ExampleUtils.java]: Thank! Will update it.

**Comment by hsaputra** [comment]
Thanks guys,  updated based comments. Will merge end of day if no more comments.

**Comment by hsaputra** [comment]
Build passes, will squash commits and merge ...

**Pull request closed without merging**